homeassistant/components/aws/notify.py

```diff
@@ -0,0 +1,278 @@
+        _in_avilable_region,
+    )
+)
+
+
+async def async_get_service(hass, config, discovery_info=None):
+    """Get the AWS notification service."""
+    import aiobotocore
+
+    session = None
+
+    if discovery_info is not None:
+        conf = discovery_info
+    else:
+        conf = config
+
+    service = conf[CONF_SERVICE]
+    region_name = conf[CONF_REGION]
+
+    aws_config = conf.copy()
+
+    del aws_config[CONF_SERVICE]
+    del aws_config[CONF_REGION]
+    if CONF_PLATFORM in aws_config:
+        del aws_config[CONF_PLATFORM]
+    if CONF_NAME in aws_config:
+        del aws_config[CONF_NAME]
+    if CONF_CONTEXT in aws_config:
+        del aws_config[CONF_CONTEXT]
+
+    if not aws_config:
+        # no platform config, use aws component config instead
+        if hass.data[DATA_SESSIONS]:
+            session = list(hass.data[DATA_SESSIONS].values())[0]
+        else:
+            raise ValueError(
+                "No available aws session for {}".format(config[CONF_NAME])
+            )
+
+    if session is None:
+        credential_name = aws_config.get(CONF_CREDENTIAL_NAME)
+        if credential_name is not None:
+            session = hass.data[DATA_SESSIONS].get(credential_name)
+            if session is None:
+                _LOGGER.warning(
+                    "No available aws session for %s", credential_name
+                )
+            del aws_config[CONF_CREDENTIAL_NAME]
+
+    if session is None:
+        profile = aws_config.get(CONF_PROFILE_NAME)
+        if profile is not None:
+            session = aiobotocore.AioSession(profile=profile, loop=hass.loop)
+            del aws_config[CONF_PROFILE_NAME]
+        else:
+            session = aiobotocore.AioSession(loop=hass.loop)
+
+    aws_config[CONF_REGION] = region_name
+
+    if service == "lambda":
+        context_str = json.dumps(
+            {"custom": conf.get(CONF_CONTEXT, {})}, cls=JSONEncoder
+        )
+        context_b64 = base64.b64encode(context_str.encode("utf-8"))
+        context = context_b64.decode("utf-8")
+        return AWSLambda(session, aws_config, context)
+
+    if service == "sns":
+        return AWSSNS(session, aws_config)
+
+    if service == "sqs":
+        return AWSSQS(session, aws_config)
+
+    raise ValueError("Unsupported service {}".format(service))
+
+
+class AWSNotify(BaseNotificationService):
+    """Implement the notification service for the AWS service."""
+
+    def __init__(self, session, aws_config):
+        """Initialize the service."""
+        self.session = session
+        self.aws_config = aws_config
+
+    def send_message(self, message, **kwargs):
+        """Send notification."""
+        raise NotImplementedError("Please call async_send_message()")
+
+    async def async_send_message(self, message="", **kwargs):
+        """Send notification."""
+        targets = kwargs.get(ATTR_TARGET)
+
+        if not targets:
+            raise HomeAssistantError("At least one target is required")
```

Comment:
We shouldn't raise. Just log error and return.

Reply:
Why not?

Reply:
It's not our standard to raise in service methods. If we know the problem there's no reason to show a stack trace.